File: gr00t/Isaac-GR00T/examples/H20/robots/sdk_package/example_py/h10-w/params/get_tcp_payload.py
import humanoid_sdk_py
import humanoid_sdk_py.h10w as h10w
import time

h10w_params = h10w.H10wParams()
time.sleep(1)

print("=== 获取 TCP 载荷参数示例 ===")

ret, tcp_payload_params = h10w_params.getTcpPayload()
if ret == 0:
    print("获取成功")
    for index, tcp_payload in tcp_payload_params:
        print("    TCP {}: {}".format(index, tcp_payload))
else:
    print("获取失败")

File: gr00t/Isaac-GR00T/examples/H20/robots/sdk_package/example/h10-w/params/get_tcp_payload.cpp
// get_tcp_payload.cpp

#include <iostream>
#include <iterator>
#include <algorithm>
#include <iomanip>
#include "h10-w/h10w_params.h"
#include "common/common_def.h"

using namespace humanoid;
using namespace humanoid::h10w;

int main() {
  std::cout << "=== 获取 TCP 载荷参数示例 ===" << std::endl;
  H10wParams params;
  TcpPayloadParams tcpPayloadParams = {
      // 留空则返回全部 TCP 载荷参数
      {1, {}},  // 获取左臂 TCP 载荷参数
      {2, {}}   // 获取右臂 TCP 载荷参数
  };

  if (!params.getTcpPayload(tcpPayloadParams)) {
    std::cout << "获取成功" << std::endl;

    for (const auto& [index, payload] : tcpPayloadParams) {
      std::cout << "    TCP " << index << ": ";
      std::copy(payload.begin(), payload.end(),
                std::ostream_iterator<double>(std::cout, " "));
      std::cout << std::endl;
    }
  } else {
    std::cout << "获取失败" << std::endl;
    return -1;
  }
  return 0;
}
